File: 1684.count-the-number-of-consistent-strings.py
#
# @lc app=leetcode id=1684 lang=python
#
# [1684] Count the Number of Consistent Strings
#

# @lc code=start
class Solution(object):
    def countConsistentStrings(self, allowed, words):
        """
        :type allowed: str
        :type words: List[str]
        :rtype: int
        """

        result = 0
        for word in words:
            if self.isGood(word, allowed):
                result += 1

        return result

# @lc code=end

File: 1684.count-the-number-of-consistent-strings.js
/*
 * @lc app=leetcode id=1684 lang=javascript
 *
 * [1684] Count the Number of Consistent Strings
 */

// @lc code=start
/**
 * @param {string} allowed
 * @param {string[]} words
 * @return {number}
 */
var countConsistentStrings = function (allowed, words) {
    const set = new Set(allowed);
    let res = 0;
    for (let i = 0; i < words.length; i++) {
        let flag = true;
        for (let j = 0; j < words[i].length; j++) {
            if (!set.has(words[i][j])) {
                flag = false;
                break;
            }
        }
        if (flag) {
            res++;
        }
    }
    return res;
};
// @lc code=end
